lib/cli-engine.js

```diff
@@ -670,15 +673,21 @@ class CLIEngine {
             // replace \ with / for Windows compatibility
             format = format.replace(/\\/g, "/");
 
+            const cwd = this.options ? this.options.cwd : process.cwd();
+
             let formatterPath;
 
             // if there's a slash, then it's a file
```

Comment:
This assumption doesn’t hold if I want to use a formatter from a scoped package on npm:

    npm install @btmills/eslint-formatter-turtles
    eslint -f @btmills/turtles ...

This raises two questions:

1. Do we want to allow formatters from scoped packages?
1. If I use a formatted in a scoped package, can I omit the `eslint-formatter-` prefix?

If the answer to both questions is yes, then we might be able to borrow the code that resolves plugins and shares configs, which handles those already. I’d find a link if I weren’t on my phone.

Reply:
Thanks. Seems like a reasonable requirement to me. Let me know if you find that link.

Reply:
https://github.com/eslint/eslint/blob/1a962e80307d3468f521ef14493a7b55fcfafece/lib/config/plugins.js#L40-L77

Reply:
Ok I've added test cases for the points raised by @btmills, but the implementation is starting to get icky. I feel like getNamespace/removeNamespace should be moved to another utility perhaps along with a more generic version of removePrefix that also received the prefix. Keen to hear your opinion on this.

Reply:
I have added test cases for the points raised by btmills and made them pass. I have also moved some helpers into a new "naming" utility (perhaps it should be called "prefix(-ing)"? The getFormatter is getting a bit complex, so I would prefer some refactoring, and maybe moving some logic to helper(s). What do you think.